src/openforms/forms/admin/tasks.py

```diff
@@ -70,6 +75,29 @@ def process_forms_export(forms_uuids: list, user_id: int) -> None:
         )
 
 
+@app.task
+def process_forms_import(import_file: str, user_id: int) -> None:
+    user = User.objects.get(id=user_id)
+    failed_files = []
+    # This deletes the temp dir once the context manager is exited
+    with tempfile.TemporaryDirectory(dir=Path(import_file).parent) as temp_dir:
+        with zipfile.ZipFile(import_file, "r") as zip_file:
```

Comment:
I'd leverage the storage class here too, following the changes in the view:

```py
from privates.storages import private_media_storage

def process_forms_import(filename: str, user_id: int) -> None:
    # This deletes the temp dir once the context manager is exited
    with tempfile.TemporaryDirectory(dir=Path(import_file).parent) as temp_dir:
        with zipfile.ZipFile(private_media_storage.open(filename), "r") as zip_file:
            ...
```

Reply:
you can pass a filename or any file-like object to `ZipFile`, which is what's happening in my suggestion. We open it explicitly using the storage, and the storage is responsible for finding it on file system (or possibly in the future in Amazon S3 or other object storages)

It's basically using the abstraction that uploaded files may not be on the same server in the future, and this code pattern can handle both situations.

Reply:
Ohh i see, i shall update my update